dash-spv/src/sync/headers/manager.rs

```diff
@@ -605,37 +582,36 @@ impl<S: StorageManager + Send + Sync + 'static, N: NetworkManager + Send + Sync
                     self.chain_state.read().await.tip_height()
                 );
 
-                    // If we failed due to missing previous header, and we're at genesis,
-                    // this might be a protocol issue where peer expects us to have genesis in compression state
-                    if matches!(e, crate::sync::headers2::ProcessError::DecompressionError(0, _))
-                        && self.chain_state.read().await.tip_height() == 0
-                    {
-                        tracing::warn!(
+                // If we failed due to missing previous header, and we're at genesis,
+                // this might be a protocol issue where peer expects us to have genesis in compression state
+                if matches!(e, crate::sync::headers2::ProcessError::DecompressionError(0, _))
```

Comment:
```suggestion
                if matches!(e, crate::sync::headers2::ProcessError::DecompressionError(0, DecompressionError::MissingPreviousHeader))
```

The comments make it sound like its some special case if previous header is missing, shouldn't it check for it then and handle other cases differently? Generally, can you maybe review/think about this error handling here a bit more, there might be some room for improvement, maybe.